File: antd-dart/example/04_files.dart
import 'package:antd/antd.dart';

/// Demonstrates file upload/download.
void main() async {
  final client = AntdClient();

  try {
    // Upload a file
    final result = await client.fileUploadPublic('/path/to/file.txt');
    print('File uploaded at ${result.address}');
    print('  storage: ${result.storageCostAtto} atto, gas: ${result.gasCostWei} wei');
    print('  chunks: ${result.chunksStored}, mode: ${result.paymentModeUsed}');

    // Download a file
    await client.fileDownloadPublic(result.address, '/path/to/output.txt');
    print('File downloaded');

    // Estimate file upload cost
    final cost = await client.fileCost('/path/to/file.txt');
    print('Estimated cost: $cost atto');
  } on AntdError catch (e) {
    print('Error: $e');
  } finally {
    client.close();
  }
}

File: antd-php/examples/04-files.php
<?php

/**
 * Example 04: Upload and download files and directories.
 *
 * Prerequisites: antd daemon running with a funded wallet.
 */

require_once __DIR__ . '/../vendor/autoload.php';

use Autonomi\Antd\AntdClient;

$client = new AntdClient();

// Estimate file upload cost
$cost = $client->fileCost('/tmp/test.txt', true, false);
echo "Estimated cost: {$cost} atto\n";

// Upload a file
$result = $client->fileUploadPublic('/tmp/test.txt');
echo "File uploaded at: {$result->address}\n";
echo "  storage: {$result->storageCostAtto} atto, gas: {$result->gasCostWei} wei\n";
echo "  chunks: {$result->chunksStored}, mode: {$result->paymentModeUsed}\n";

// Download the file
$client->fileDownloadPublic($result->address, '/tmp/downloaded.txt');
echo "File downloaded to /tmp/downloaded.txt\n";

